opencolorio_config_aces/clf/transforms/arri/generate.py

```diff
@@ -0,0 +1,450 @@
+    """
+    Generate the collection of LogC4 transforms.
+
+    Parameters
+    ----------
+    transforms_set : dict
+        A `dict` of transforms to append to.
+    output_directory : PosixPath or WindowsPath
+        Directory where generated CLF files should be saved.
+
+    Returns
+    -------
+    dict
+        Dictionary of *OpenColorIO* `Transform` instances.
+    """
+
+    # Generate ARRI LogC4 to ACES 2065-1 Transform
+    name = "ARRI_LogC4_to_ACES2065-1"
+    aces_id = "urn:ampas:aces:transformId:v1.5:IDT.ARRI.LogC4.a1.v1"
+    input_descriptor = "ARRI LogC4"
+    output_descriptor = "ACES2065-1"
+    clf_transform_id = format_clf_transform_id(FAMILY, GENUS, name, VERSION)
+    filename = output_directory / clf_basename(clf_transform_id)
+    transforms_set[filename] = generate_clf_transform(
+        filename,
+        [_build_logc4_curve(), _build_awg4_mtx()],
+        clf_transform_id,
+        f"{input_descriptor} to {output_descriptor}",
+        input_descriptor,
+        output_descriptor,
+        aces_id,
+    )
+
+    # Generate ARRI LogC4 Curve Transform
+    name = "ARRI_LogC4_Curve_to_Linear"
+    input_descriptor = "ARRI LogC4 Curve"
+    output_descriptor = "Relative Scene Linear"
+    clf_transform_id = format_clf_transform_id(FAMILY, GENUS, name, VERSION)
+    filename = output_directory / clf_basename(clf_transform_id)
+    transforms_set[filename] = generate_clf_transform(
+        filename,
+        [_build_logc4_curve()],
+        clf_transform_id,
+        f"{input_descriptor} to {output_descriptor}",
+        input_descriptor,
+        output_descriptor,
+    )
+
+    # Generate Linear ARRI Wide Gamut 4 to ACES 2065-1 Transform
+    name = "Linear_ARRI_Wide_Gamut_4_to_ACES2065-1"
+    input_descriptor = "Linear ARRI Wide Gamut 4"
+    output_descriptor = "ACES2065-1"
+    clf_transform_id = format_clf_transform_id(FAMILY, GENUS, name, VERSION)
+    filename = output_directory / clf_basename(clf_transform_id)
+    transforms_set[filename] = generate_clf_transform(
+        filename,
+        [_build_awg4_mtx()],
+        clf_transform_id,
+        f"{input_descriptor} to {output_descriptor}",
+        input_descriptor,
+        output_descriptor,
+    )
+
+    return transforms_set
+
+
+def generate_clf_arri(output_directory):
+    """
+    Generate CLF files for ARRI color encodings and save to disk.
+
+    Parameters
+    ----------
+    output_directory : PosixPath or WindowsPath
+        Directory where generated CLF files should be saved.
+
+    Returns
+    -------
+    dict
+        Dictionary of *OpenColorIO* `Transform` instances.
+
+    References
+    ----------
+    -
+
+    """
+
+    output_directory.mkdir(parents=True, exist_ok=True)
+
+    clf_transforms = {}
+    clf_transforms = _generate_logc3_transforms(
+        clf_transforms, output_directory, ei_list=[800]
+    )
+    clf_transforms = _generate_logc4_transforms(
+        clf_transforms, output_directory
+    )
```

Comment:
This is a bit anti-pattern, following-up the entry-point, i.e. `_main`, we get into this definition `generate_clf_arri`, there we can see the `_generate_logc3_transforms` and  `_generate_logc4_transforms` assigning to the same variable.

If you don't look at the signature of the `_generate_logc*_transforms` definitions you would think that this is a mistake.

Ideally the definitions should either

1. Update the passed variable in place, in which case they should not return anything.
2. Not take an argument and return the generated transforms.

My preference would be for 2 because:

- This makes the definitions pure.
- It makes the usage simpler: We could remove the weirdly name `transforms_set` argument which is actually not a `set` but a `dict`.
- It is similar to other CLF definitions.

Happy to merge and do those changes in a subsequent PR!

Reply:
Agreed. Simply didn't think of a nice way to merge dicts.

Fixed.

Reply:
Awesome, thank you!